### PR TITLE
Extending SpriteFrameCache to read PolygonInfo from atlas plist file

### PR DESCRIPTION
**Pull request by CodeAndWeb**
Extended SpriteFrameCache to read polygon info from atlas .plist file.
The polygon information is added to a sprite instead of quads when creating a sprite from a named sprite frame.
Polygon creation can be done offline using TexturePacker 4.0.0.
Works with auto batching.

### CODE REVIEW

**Comment by ricardoquesada** [comment]
PR looks good to me.

File: cocos/2d/CCSpriteFrameCache.h [hunk 1]
```diff
@@ -223,6 +249,16 @@ class CC_DLL SpriteFrameCache : public Ref
     */
     void removeSpriteFramesFromDictionary(ValueMap& dictionary);
 
+    /** Parses list of space-separated integers */
+    void parseIntegerList(const std::string string, std::vector<int> &res);
```
Comment (zilongshanren) [inline cocos/2d/CCSpriteFrameCache.h]: `const std::string string` should be `const std::string& string`?

File: cocos/2d/CCSpriteFrameCache.cpp [hunk 2]
```diff
@@ -79,6 +80,63 @@ SpriteFrameCache::~SpriteFrameCache()
...
+                                             const std::vector<int> &verticesUV,
+                                             const std::vector<int> &triangleIndices,
+                                             PolygonInfo &info)
+{
+    size_t vertexCount = vertices.size();
+    size_t indexCount = triangleIndices.size();
+    
+    float scaleFactor = CC_CONTENT_SCALE_FACTOR();
+
+    V3F_C4B_T2F *vertexData = new V3F_C4B_T2F[vertexCount];
```
Comment (zilongshanren) [inline cocos/2d/CCSpriteFrameCache.cpp]: when does the structure memory of `vertexData` got released? I think it's the polygonInfo's responsibility to do the house clean work. So we should set `_isOwner` to true inside polygon info, thoughts?

**Comment by zilongshanren** [comment]
@pandamicro  
I'm OK with this PR now.

**Comment by zilongshanren** [comment]
[ci rebuild]

**Comment by pandamicro** [comment]
@CodeAndWeb  Thanks for supporting polygon sprite in TexturePacker and thanks for bringing this to cocos2d too.

**Comment by CodeAndWeb** [comment]
@pandamicro Thanks for the merge.

**Pull request merged**